### PR TITLE
Fix doc build on catalog API

### PR DESCRIPTION
**Pull request by alamb**
This targets the catalog API PR in DataFusion, https://github.com/apache/datafusion/pull/11516

I broke the doc tests in  https://github.com/findepi/datafusion/pull/4 due a broken link

This PR fixes that (and improves the example slightly)

### CODE REVIEW

File: datafusion/catalog/src/session.rs [hunk 1]
```diff
@@ -52,7 +52,7 @@ use std::sync::Arc;
 /// ```
 /// # use datafusion_catalog::Session;
 /// # use datafusion_common::{Result, exec_datafusion_err};
-/// struct SessionState {}
+/// # struct SessionState {}
```
Comment (alamb) [inline datafusion/catalog/src/session.rs:55]: Putting a `#` in front of this means the statement is not shown in the resulting documentation

File: datafusion/catalog/src/session.rs [hunk 2]
```diff
@@ -63,7 +63,7 @@ use std::sync::Arc;
 /// ```
 ///
 /// [`SessionState`]: https://docs.rs/datafusion/latest/datafusion/execution/session_state/struct.SessionState.html
-/// [TableProvider]: crate::TableProvider
+/// [`TableProvider`]: crate::TableProvider
```
Comment (alamb) [inline datafusion/catalog/src/session.rs:66]: This is the fix for the doc failure 🤦

**Comment by alamb** [comment]
cc @findepi

File: datafusion/catalog/src/session.rs (hunk 1 above)
Comment (findepi) [reply datafusion/catalog/src/session.rs:55]: thanks for the explanation! ❤️

**Comment by findepi** [comment]
thank you!

**Pull request merged**